yarn-project/archiver/src/archiver/archiver.ts

```diff
@@ -109,11 +113,23 @@ export class Archiver implements ArchiveSource {
       pollingInterval: config.viemPollingIntervalMS,
     });
 
+    // TODO(#4492): Nuke this once the old inbox is purged
+    let newInboxAddress!: EthAddress;
+    {
+      const rollup = getContract({
+        address: getAddress(config.l1Contracts.rollupAddress.toString()),
+        abi: RollupAbi,
+        client: publicClient,
+      });
+      newInboxAddress = EthAddress.fromString(await rollup.read.NEW_INBOX());
+    }
```

Comment:
This ^ is the least annoying way of getting the new inbox address. Once the old inbox is purged the new inbox address will be passed config.l1Contracts.inboxAddress as well.

BTW I feel like having all these addresses passed via config is quite shit and it would be better to just pass registry address and fetch the rest from chain.